src/mutation/RelayMutation.js

```diff
@@ -50,12 +50,29 @@ class RelayMutation<Tp: Object> {
     route: RelayMetaRoute
   ) => Variables;
 
-  props: Tp;
+  _context: RelayContext;
   _didShowFakeDataWarning: boolean;
+  _unresolvedProps: Tp;
+  props: Tp;
 
   constructor(props: Tp) {
-    this._didShowFakeDataWarning = false;
-    this._resolveProps(props);
+    this._unresolvedProps = props;
+  }
+
+  /**
+   * @internal
+   */
+  bindContext(context: RelayContext): void {
+    if (!this._context) {
+      this._context = context;
+      this._resolveProps();
```

Comment:
> We could probably even write a codemod that would move existing constructor logic after the `super` call into the callback method.

@josephsavona, I've never written any codemod before, but I could try if it would help to merge this PR sooner. Should I do it?